### PR TITLE
Replace new-pass-manager-fast builder with llvm-dylib-all

### PR DESCRIPTION
**Pull request by tbaederr**
Test a LLVM build with the dynamic library enabled. Remove the
clang-x86_64-debian-new-pass-manager-fast builder since the new pass
manager has been the default since Feb 21.

I've modeled this after the existing `x86_64-fedora-clang` builder.

### CODE REVIEW

File: buildbot/osuosl/master/config/builders.py
Comment (gribozavr) [inline buildbot/osuosl/master/config/builders.py:1828]:
```suggestion
    {'name' : 'llvm-x86_64-debian-dylib',
```

I'm not quite certain what "all" means.

File: buildbot/osuosl/master/config/builders.py
Comment (gribozavr) [inline buildbot/osuosl/master/config/builders.py:1846]: Please remove the CCACHE flags and DCMAKE_C_COMPILER. Instead, please imitate the configuration of the builder that you're replacing (environment variables).

File: buildbot/osuosl/master/config/builders.py
Comment (tbaederr) [reply buildbot/osuosl/master/config/builders.py:1828]: I was only building the `x86_64` target previously and changed to build all targets, thus the "all" in the name.

File: buildbot/osuosl/master/config/builders.py
Comment (gribozavr) [inline buildbot/osuosl/master/config/builders.py:1850]: Please include the PATH as well.

Also, any reason to use gcc instead of clang as the host compiler?

Comment (tbaederr) [reply buildbot/osuosl/master/config/builders.py:1850]: We just discussed this and it is not a requirement for the test, I'll change it to clang.

**Comment by tstellar** [comment]
Could we also have the same configuration added for the release branch?

**Review by gribozavr — APPROVED** [review]

**Comment by tstellar** [comment]
@tbaederr We can always add the release branch configuration in a follow up commit, so no need to hold up this patch for it.

**Comment by gribozavr** [comment]
@tbaederr Do you have commit access or do you want me to merge it?

**Comment by tbaederr** [comment]
I can do it, thanks

**Pull request merged**
